client/src/features/display/displaySlice.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright 2023 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { TypedUseSelectorHook, useSelector } from "react-redux";
+import { Display, ProjectConfig } from "./display";
+
+const initialState: Display = {
+  modals: {
+    ssh: {
+      show: false,
+      projectPath: "",
+      gitUrl: ""
+    }
+  }
+};
+
+export const displaySlice = createSlice({
+  name: "display",
+  initialState,
+  reducers: {
+    showSshModal: (state, action: PayloadAction<ProjectConfig>) => {
+      return {
+        ...state, modals: {
+          ...state.modals, ssh: {
+            ...state.modals.ssh,
+            show: true,
+            projectPath: action.payload.projectPath,
+            gitUrl: action.payload.gitUrl,
+          }
```

Comment:
This kind of thing isn't necessary. The redux-toolkit reducer arguments are [wrapped using Immer](https://redux-toolkit.js.org/usage/immer-reducers) so you can just mutate the values.

```suggestion
      state.modals.show = true;
      state.modals.projectPath = action.payload.projectPath;
      state.modals.gitUrl = action.payload.gitUrl;
```

Reply:
Oh nice! That simplifies reading the code. I'll add this and the following changes (I'll add another commit since I need to add `.ssh` and I'll use a single object to further simplify)